### ISSUE TITLE
Bug: RegisterDeviceNotificationA returns `*mut c_void` in Rust and not `HDEVNOTIFY `

### ISSUE THREAD

**Issue by levigross** [microsoft/win32metadata#1514]
Hello :) 

I think I found another discrepancy between the API listed on MSDN (c++) and the rust crate using with the function `RegisterDeviceNotificationA`.

In Rust the function is defined as:

```
pub unsafe fn RegisterDeviceNotificationA<P0>(
    hrecipient: P0,
    notificationfilter: *const c_void,
    flags: REGISTER_NOTIFICATION_FLAGS
) -> *mut c_voidwhere
    P0: IntoParam<HANDLE>,
```
Src: https://microsoft.github.io/windows-docs-rs/doc/windows/Win32/UI/WindowsAndMessaging/fn.RegisterDeviceNotificationA.html 

In C++ it is defined as:
```
HDEVNOTIFY RegisterDeviceNotificationA(
  [in] HANDLE hRecipient,
  [in] LPVOID NotificationFilter,
  [in] DWORD  Flags
);
```
Src: https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-registerdevicenotificationa 

In short, the Rust doesn't have the `HDEVNOTIFY ` type
